src/main/java/com/firebolt/jdbc/connection/FireboltConnectionUserPasswordAuthentication.java

```diff
@@ -0,0 +1,80 @@
+import com.firebolt.jdbc.client.authentication.OldServiceAccountAuthenticationRequest;
+import com.firebolt.jdbc.client.authentication.UsernamePasswordAuthenticationRequest;
+import com.firebolt.jdbc.connection.settings.FireboltProperties;
+import com.firebolt.jdbc.service.FireboltAuthenticationService;
+import com.firebolt.jdbc.service.FireboltEngineApiService;
+import com.firebolt.jdbc.service.FireboltEngineInformationSchemaService;
+import com.firebolt.jdbc.service.FireboltEngineService;
+import com.firebolt.jdbc.service.FireboltStatementService;
+import lombok.NonNull;
+import okhttp3.OkHttpClient;
+import org.apache.commons.lang3.StringUtils;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class FireboltConnectionUserPasswordAuthentication extends FireboltConnection {
+    private final FireboltEngineService fireboltEngineService;
+
+    FireboltConnectionUserPasswordAuthentication(@NonNull String url,
+                                                 Properties connectionSettings,
+                                                 FireboltAuthenticationService fireboltAuthenticationService,
+                                                 FireboltStatementService fireboltStatementService,
+                                                 FireboltEngineInformationSchemaService fireboltEngineService) throws SQLException {
+        super(url, connectionSettings, fireboltAuthenticationService, fireboltStatementService);
+        this.fireboltEngineService = fireboltEngineService;
+        connect();
+    }
+
+    @ExcludeFromJacocoGeneratedReport
+    FireboltConnectionUserPasswordAuthentication(@NonNull String url, Properties connectionSettings) throws SQLException {
+        super(url, connectionSettings);
+        OkHttpClient httpClient = getHttpClient(loginProperties);
+        ObjectMapper objectMapper = FireboltObjectMapper.getInstance();
+        this.fireboltEngineService = new FireboltEngineApiService(new FireboltAccountClient(httpClient, objectMapper, this, loginProperties.getUserDrivers(), loginProperties.getUserClients()));
+        connect();
+    }
+
+    @Override
+    protected void authenticate() throws SQLException {
+        String accessToken = getAccessToken(loginProperties).orElse(StringUtils.EMPTY);
+        FireboltProperties propertiesWithAccessToken = loginProperties.toBuilder().accessToken(accessToken).build();
+        Engine engine = fireboltEngineService.getEngine(propertiesWithAccessToken);
+        this.sessionProperties = loginProperties.toBuilder().host(engine.getEndpoint()).database(engine.getDatabase()).engine(engine.getName()).build();
+    }
+
+    @Override
+    protected FireboltProperties extractFireboltProperties(String jdbcUri, Properties connectionProperties) {
+        FireboltProperties properties = super.extractFireboltProperties(jdbcUri, connectionProperties);
+        boolean systemEngine = "system".equals(properties.getEngine());
```

Comment:
Cannot because this constant has been removed from branch with new identity as redundant. However you are right that using constant is better, so I have restored it but put it into more relevant now place - `FireboltConnectionUserPassword`